src/microbe_stage/MicrobeHUD.cs

```diff
@@ -779,19 +783,27 @@ private void UpdateATP(float delta)
             return;
 
         var atpAmount = 0.0f;
-        var capacity = 4.0f;
 
+        // Update to the player's current ATP, unless the player is dead
         if (stage.Player != null)
         {
             var compounds = GetPlayerColonyOrPlayerStorage();
 
-            atpAmount = Mathf.Ceil(compounds.GetCompoundAmount(atp));
-            capacity = compounds.Capacity;
+            atpAmount = compounds.GetCompoundAmount(atp);
+            maxATP = compounds.Capacity;
+        }
+
+        atpBar.MaxValue = maxATP * 10.0f;
+
+        // If the current ATP is close to full, just pretend that it is to keep the bar from flickering
+        if (maxATP - atpAmount < Math.Max(maxATP / 20.0f, 0.1f))
+        {
+            atpAmount = maxATP;
         }
 
-        atpBar.MaxValue = capacity;
-        atpBar.Value = MathUtils.Lerp((float)atpBar.Value, atpAmount, 3.0f * delta, 0.1f);
-        atpLabel.Text = StringUtils.FormatNumber(atpAmount) + " / " + StringUtils.FormatNumber(capacity);
+        GUICommon.SmoothlyUpdateBar(atpBar, atpAmount * 10.0f, delta);
+        atpLabel.Text = atpAmount.ToString("F1", CultureInfo.CurrentCulture) + " / "
+            + maxATP.ToString("F1", CultureInfo.CurrentCulture);
```

Comment:
why does ATP use `.ToString("F1"` but HP uses `StringUtils.FormatNumber`?

Reply:
Because ATP can be fractional, while HP can't. Using F1 would add a meaningless ".0" to all HP numbers.

Reply:
What about `F0` then? (I think that might work)